### PR TITLE
Initial parts of design doc for parallel commit

### PR DESCRIPTION
**Pull request by nrc**
PTAL @MyonKeminta (feel free to add sections later, I don't intend for this to be a final layout).

### CODE REVIEW

**Comment by MyonKeminta** [comment]
Fine. Should I push my content directly to this PR?

**Comment by youjiali1995** [comment]
> Fine. Should I push my content directly to this PR?

I think it's ok.

**Comment by MyonKeminta** [comment]
@youjiali1995 Is this ok?  Is there anything else I need to add?

File: design/parallel-commit/parallel-commit-known-issues-and-solutions.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+4. Continue performing prewrite and persist `T2` to the locks written down to engine.
+
+### Some other ideas
+
+* Make something like HLC, which doesn't seem to be practical for us.
+* ...
+
+## Replica Read (By @gengliqi)
+
+In the solutions to the locking problem, writing are performed on leaders, and it needs to know the `max_read_ts`. However if follower read read is enabled, the leader need to know the `max_read_ts` among all replicas of the Region by some way, and reading on followers should be blocked when the leader has an ongoing conflicting prewrite. Here is one possible solution to this problem.
```
Comment (youjiali1995) [inline design/parallel-commit/parallel-commit-known-issues-and-solutions.md:66]:
```suggestion
In the solutions to the locking problem, writing are performed on leaders, and it needs to know the `max_read_ts`. However if follower read is enabled, the leader need to know the `max_read_ts` among all replicas of the Region by some way, and reading on followers should be blocked when the leader has an ongoing conflicting prewrite. Here is one possible solution to this problem.
```

File: design/parallel-commit/parallel-commit-known-issues-and-solutions.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+2. Lock the memlock and acquire latches, saving `T1` to them.
+3. Atomically get the `max_read_ts` again as `T2`.
+4. Continue performing prewrite and persist `T2` to the locks written down to engine.
+
+### Some other ideas
+
+* Make something like HLC, which doesn't seem to be practical for us.
+* ...
+
+## Replica Read (By @gengliqi)
```
Comment (youjiali1995) [inline design/parallel-commit/parallel-commit-known-issues-and-solutions.md:64]: Another solution is `ReadIndex` carries read_ts and key range and treat it as normal read operations.

**Comment by youjiali1995** [comment]
> @youjiali1995 Is this ok? Is there anything else I need to add?

Yes! It's time to create a project for it and break down it. @nrc

File: design/parallel-commit/initial-design.md [hunk 3]
```diff
@@ -1 +1,85 @@
...
+## Staffing
+
+The following people are available for work on this project (as of 2020-06-15):
+
+* Zhenjing (@MyonKeminta): minimal time until CDC project is complete
+* Zhaolei (@youjiali1995): review + minimal time
+* Nick (@nrc): approximately full time
+* Yilin (@sticnarf): ???
+* Fangsong (@longfangsong): approximately full time after apx one month (although work to be finalised)
+* Rui Xu (@cfzjywxk): ???
```
Comment (nrc) [inline design/parallel-commit/initial-design.md:67]: @sticnarf @cfzjywxk how much time (if any) do you think you will have to work on parallel commit?

Comment (cfzjywxk) [reply design/parallel-commit/initial-design.md:67]: @nrc 
Maybe sometime(approximately 1/3) before another [requirement](https://internal.pingcap.net/jira/browse/PRM-402)  is done.

**Pull request merged**